WrightTools/data/_join.py

```diff
@@ -93,45 +148,122 @@ def from_dict(d, parent=None):
         if variable_name not in vs.keys():
-            shape = tuple(
-                1 if i == 1 else n for i, n in zip(datas[0][variable_name].shape, out.shape)
-            )
             # **attrs passes the name and units as well
-            out.create_variable(shape=shape, **datas[0][variable_name].attrs)
-    # channels
+            out.create_variable(
+                shape=get_shape(out, datas, variable_name),
+                **datas[0][variable_name].attrs,
+                dtype=datas[0][variable_name].dtype
+            )
+            count[variable_name] = np.zeros_like(out[variable_name], dtype=int)
+
+    def combine(data, out, item_name, new_idx, transpose, slice_):
+        old = data[item_name]
+        new = out[item_name]
+        vals = np.empty_like(new)
+        # Default fill value based on whether dtype is floating or not
+        if vals.dtype.kind in "fcmM":
+            vals[:] = np.nan
+        else:
+            vals[:] = 0
+        # Use advanced indexing to populate vals, a temporary array with same shape as out
+        valid_index = tuple(wt_kit.valid_index(new_idx, new.shape))
+        vals[valid_index] = old[:].transpose(transpose)[slice_]
+
+        # Overlap methods are accomplished by adding the existing array with the one added
+        # for this particular data. Thus locations which should be set, but conflict by
+        # the method chosen are set to 0. Handling for floating point vs. integer types may vary.
+        # For floating types, nan indicates invalid, and must be explicitly allowed to add in.
+        if method == "first":
+            # Set any locations which have already been populated
+            vals[~np.isnan(new[:])] = 0
+            if not vals.dtype.kind in "fcmM":
+                vals[count[item_name] > 0] = 0
+        elif method == "last":
+            # Reset points which are to be overwritten
+            new[~np.isnan(vals)] = 0
+            if not vals.dtype.kind in "fcmM":
+                new[valid_index] = 0
+        elif method == "min":
+            rep_new = new > vals
+            rep_vals = vals > new
+            new[rep_new] = 0
+            vals[rep_vals] = 0
+        elif method == "max":
+            rep_new = new < vals
+            rep_vals = vals < new
+            new[rep_new] = 0
+            vals[rep_vals] = 0
+        # Ensure that previously NaN points which have values are written
+        new[np.isnan(new) & ~np.isnan(vals)] = 0
+        # Ensure that new data does not overwrite any previous data with nan
+        vals[np.isnan(vals)] = 0
+        # Track how many times each point is set (for mean)
+        count[item_name][valid_index] += 1
+        new[:] += vals
+
     for data in datas:
         new_idx = []
+        transpose = []
+        slice_ = []
         for variable_name in vs.keys():
-            p = data[variable_name][:][np.newaxis, ...]
+            # p is at most 1-D by precondition to join
```

Comment:
Where is this tested for, and what exception is raised otherwise?

Reply:
https://github.com/wright-group/WrightTools/blob/e7f14ed110301d3f440b68ffce014adadb2f8752/WrightTools/data/_join.py#L119

It is actually enforced by the structure of the `from_dict` It will not raise an error, but it will flatten the array (and combine with atol/rtol)

Reply:
So you CAN hand join multidimensional axes, but you will get 1D out